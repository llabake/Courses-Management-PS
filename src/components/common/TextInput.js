import React from 'react';
import {
  string,
  func,
} from 'prop-types';

const TextInput = ({
  name, label, onChange, placeholder, value, error, 
}) => {
  let wrapperClass = 'form-group';
  if (error && error.length) {
    wrapperClass = `${wrapperClass} has-error`;
  }
  return (
    <div className={wrapperClass}>
      <label htmlFor={name}>{label}</label>
      {
        error && <div className="alert alert-danger">{error}</div>
      }
      <div className="field">
        <input
          type="text"
          name={name}
          className="form-control"
          placeholder={placeholder}
          value={value}
          onChange={onChange}
        />
      </div>
    </div>
  );
};

export default TextInput;
