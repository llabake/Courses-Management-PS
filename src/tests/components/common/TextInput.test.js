import React from 'react';
import { mount } from 'enzyme';
import TextInput from "../../../components/common/TextInput";

describe('<TextInput />', () => {
  it('field component renders correctly', () => {
    const props = {
      label: '',
      placeholder: '',
      name: '',
      error: [],
      value: '',
      onChange: jest.fn(),
    };
    const wrapper = mount(<TextInput {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('Text input field component show error when they occur', () => {
    const props = {
      label: 'Title',
      placeholder: '',
      name: 'title',
      error: ['Course title must be at least 2 characters'],
      value: 'e',
      onChange: jest.fn(),
    };
    const wrapper = mount(<TextInput {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('.has-error').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
