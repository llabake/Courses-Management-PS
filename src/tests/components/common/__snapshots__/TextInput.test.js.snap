// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Text input field component show error when they occur 1`] = `
<TextInput
  error={
    Array [
      "Course title must be at least 2 characters",
    ]
  }
  label="Title"
  name="title"
  onChange={[MockFunction]}
  placeholder=""
  value="e"
>
  <div
    className="form-group has-error"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="alert alert-danger"
    >
      Course title must be at least 2 characters
    </div>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[MockFunction]}
        placeholder=""
        type="text"
        value="e"
      />
    </div>
  </div>
</TextInput>
`;

exports[`<TextInput /> field component renders correctly 1`] = `
<TextInput
  error={Array []}
  label=""
  name=""
  onChange={[MockFunction]}
  placeholder=""
  value=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="alert alert-danger"
    />
    <div
      className="field"
    >
      <input
        className="form-control"
        name=""
        onChange={[MockFunction]}
        placeholder=""
        type="text"
        value=""
      />
    </div>
  </div>
</TextInput>
`;
